docs/src/rules/logical-assignment-operators.md

```diff
@@ -0,0 +1,122 @@
+---
+title: logical-assignment-operators
+layout: doc
+edit_link: https://github.com/eslint/eslint/edit/main/docs/src/rules/logical-assignment-operators.md
+rule_type: suggestion
+---
+
+
+
+Require or disallow assignment logical operator shorthand
+
+## Rule Details
+
+This rule requires or disallows logical assignment operator shorthand.  
+ES2021 introduces the assignment operator shorthand for the logical operators `||`, `&&` and `??`.
+Before this was only allowed for mathematical operations such as `+` or `*` (see the rule [operator-assignment](./operator-assignment)).
+The shorthand can be used if the assignment target and the left expression of a logical expression are the same.
+For example `a = a || b` can be shortened to `a ||= b`.
+
+### Options
+
+This rule has a string and an object option.
+String option:
+
+* `"always"` (default)
+* `"never"`
+
+Object option (only available if string option is set to `"always"`):
+
+* `"enforceForIfStatements": false`(default) Do *not* check for equivalent if statements
+
+* `"enforceForIfStatements": true` Check for equivalent if statements
+
+#### always
+
+Examples of **incorrect** code for this rule with the default `"always"` option:
+
+::: incorrect
+
+```js
+/*eslint logical-assignment-operators: ["error", "always"]*/
+
+a = a || b
+a = a && b
+a = a ?? b
+```
+
+:::
+
+Examples of **correct** code for this rule with the default `"always"` option:
+
+::: correct
+
+```js
+/*eslint logical-assignment-operators: ["error", "always"]*/
+
+a = b
+a ||= b
+a += b
+```
+
+:::
+
+#### never
+
+Examples of **incorrect** code for this rule with the `"never"` option:
+
+::: incorrect
+
+```js
+/*eslint logical-assignment-operators: ["error", "never"]*/
+
+a ||= b
+a &&= b
+```
+
+:::
+
+Examples of **correct** code for this rule with the `"never"` option:
+
+::: correct
+
+```js
+/*eslint logical-assignment-operators: ["error", "never"]*/
+
+a = a || b
+a = a ?? b
+```
+
+:::
+
+#### enforceForIfStatements
+
+This option checks for additional patterns with if statements which could be expressed with the logical assignment operator.
+::: correct
+
+```js
+/*eslint logical-assignment-operators: ["error", "always", { enforceForIfStatements: true }]*/
+
+if (a) b = c
+if (a === 0) a = b
+```
+
+:::
+
+::: incorrect
+
+```js
```

Comment:
Also missing text "Examples for...". Also, can we place incorrect examples before correct examples, for consistency with other sections.